cpp/finite-element.cpp

```diff
@@ -159,6 +156,30 @@ const std::vector<std::vector<int>>& FiniteElement::entity_dofs() const
 //-----------------------------------------------------------------------------
 std::vector<Eigen::ArrayXXd>
 FiniteElement::tabulate(int nd, const Eigen::ArrayXXd& x) const
+{
+  const int tdim = cell::topological_dimension(_cell_type);
+  int ndsize = 1;
+  for (int i = 1; i <= nd; ++i)
+    ndsize *= (tdim + i);
+  for (int i = 1; i <= nd; ++i)
+    ndsize /= i;
+
+  const int ndofs = _coeffs.rows();
+  const int vs = value_size();
+
+  std::vector<double> basis_data(ndsize * x.rows() * ndofs * vs);
```

Comment:
Can we avoid this dynamic allocation?

Reply:
This version of tabulate is only used by the Python wrapper. The c-interface uses `tabulate_to_memory`